src/main/java/com/thealgorithms/maths/Fibonacci.java

```diff
@@ -0,0 +1,41 @@
+package com.thealgorithms.maths;
+
+import java.math.BigInteger;
+
+/**
+ * This class provides methods for calculating Fibonacci numbers using BigInteger for large values of 'n'.
+ */
+public final class Fibonacci {
+
+    private Fibonacci() {
+        // Private constructor to prevent instantiation of this utility class.
+    }
+
+    /**
+     * Calculates the nth Fibonacci number.
+     *
+     * @param n The index of the Fibonacci number to calculate.
+     * @return The nth Fibonacci number as a BigInteger.
+     * @throws IllegalArgumentException if the input 'n' is a negative integer.
+     */
+    public static BigInteger calFibcompute(final int n) {
```

Comment:
Please use the same method name as in the other class. Something like `compute` or `calculate`.